src/dune_rules/scope.ml

```diff
@@ -85,13 +85,13 @@ module DB = struct
       else Lib.DB.Resolve_result.not_found
     | Found lib -> Memo.return (Lib.DB.Resolve_result.found lib)
     | Deprecated_library_name lib ->
-      Memo.return (Lib.DB.Resolve_result.redirect_in_the_same_db lib)
+      Memo.return (Lib.DB.Resolve_result.redirect parent lib)
```

Comment:
Is `parent` always the public library database here?

Reply:
yes. I'll move these functions more closely together and make that explicit in a comment

Reply:
it adds a bit of whitespace diff noise, but I did the above in https://github.com/ocaml/dune/pull/10354/commits/3e730834ed182f5bfaed4cc16c904abd23f795c7